packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -432,11 +432,20 @@ class _MenuAnchorState extends State<MenuAnchor> {
       );
     }
 
-    return _MenuAnchorScope(
-      anchorKey: _anchorKey,
-      anchor: this,
-      isOpen: _isOpen,
-      child: child,
+    // This `Shortcuts` is needed so that shortcuts work when the focus is on
+    // MenuAnchor (specifically, the root menu, since submenus have their own
+    // `Shortcuts`.)
+    return Shortcuts(
```

Comment:
Since this only affects web, would it make sense to only include this widget if kIsWeb is true? It adds an extra focus node to the tree that wouldn't need to be there on other platforms.

Reply:
Yeah I understand the concern. But how I actually phrased the unit test is that "`MenuAnchor`'s shortcuts should work without relying on `WidgetsApp.shortcuts`". I think even if `WidgetsApp.shortcuts` is changed and arrow keys are repurposed to other functions globally, when you focus on a `MenuAnchor`, your arrow keys should still navigate the menu, independently from platforms. Therefore I think this change should apply to all platforms.

Reply:
SGTM